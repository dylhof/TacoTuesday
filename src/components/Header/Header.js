import React, { Component } from 'react';
import  Menu  from '../Menu/Menu';

export class Header extends Component {
  state = {
    showMenu: false
  }

  toggleShowMenu = () => {
    this.setState({ showMenu: !this.state.showMenu })
  }

  render() {
    return (
      <header className='header'>
      {!this.state.showMenu && 
         <i className="fas fa-chevron-right fa-2x" onClick={this.toggleShowMenu}></i>}
        <h1>The Taste Of Taco Tuesday</h1>
        {this.state.showMenu && <Menu toggleShowMenu={this.toggleShowMenu} showMenu={this.state.showMenu}/>}
      </header>
    )
  }
}