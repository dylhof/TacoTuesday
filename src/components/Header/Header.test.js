import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header', () => {
  let wrapper;
  beforeEach(() =>{
    wrapper = shallow(<Header/>);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should toggle the showMenu property in state', () => {
    expect(wrapper.state('showMenu')).toEqual(false);
    wrapper.instance().toggleShowMenu();
    expect(wrapper.state('showMenu')).toEqual(true);
  })
});