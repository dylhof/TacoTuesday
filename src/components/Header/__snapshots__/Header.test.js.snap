// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot 1`] = `
<header
  className="header"
>
  <i
    class="fas fa-chevron-right fa-2x"
    onClick={[Function]}
  />
  <h1>
    The Taste Of Taco Tuesday
  </h1>
</header>
`;
